src/Maestro/Maestro.AzureDevOps/IAzureDevOpsTokenProvider.cs

```diff
@@ -2,14 +2,30 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Text.RegularExpressions;
 using System.Threading.Tasks;
 
 namespace Maestro.AzureDevOps
 {
     public interface IAzureDevOpsTokenProvider
     {
-        Task<string> GetTokenForRepository(string repositoryUrl);
+        Task<string> GetTokenForAccount(string account);
+    }
+
+    public static class AzureDevOpsTokenProviderExtensions
+    {
+        private static readonly Regex AccountNameRegex = new Regex(@"^https://dev\.azure\.com/(?<account>[a-zA-Z0-9]+)/");
```

Comment:
>[a-zA-Z0-9]+ [](start = 105, length = 12)

You could probably do "not a slash" here... but I'm pretty sure only "devdiv" "mseng" and "dnceng" are the only values we'll ever match, so it doesn't really matter. :-)